modules/phase_field/test/tests/grain_growth/hex.i

```diff
@@ -29,6 +29,7 @@
     coloring_algorithm = bt
     x_offset = .5
     grain_num = 4
+    patch_size = 4
```

Comment:
This is modifying an example which previously used a linear search to find a closest grain point, but if you now use patch_size=4 to find the lowest-id-among-the-closest-4-grain-points, shouldn't the results have changed?

(I'm still not clear on why the patch_size != 1 case is worth supporting at all, so perhaps I'm missing something here)

Reply:
Yes, I am trying to avoid regolding tests. In this particular example, for a given point, there might be multiple solutions. The naive linear algorithm takes the grain with smallest ID as the solution.

There are several options for me right now, but I do not which one is the best

1) Use patch size, so we can pick the grain with smallest ID. `patch_size` might be useful for getting rid of randomness of certain causes. 

2) Do not use KDTree at all for this example (use_kdtree=false). The example does not have many grains, but users can start from the example, and use more and more grains

3) Regold tests. I normally avoid regained test

4) It is not really related to this example. It might be a good idea to use the linear algorithm when the number of grains is small.

Reply:
> use patch_size=4 to find the lowest-id-among-the-closest-4-grain-points

We consider a smaller ID only when the distances are the same